src/main/java/com/hubspot/jinjava/lib/filter/TitleFilter.java

```diff
@@ -30,10 +29,36 @@ public String getName() {
 
   @Override
   public Object filter(Object var, JinjavaInterpreter interpreter, String... args) {
-    if (var instanceof String) {
-      String value = (String) var;
-      return WordUtils.capitalize(value.toLowerCase());
+    if (var == null) {
+      return null;
     }
-    return var;
+
+    String value = var.toString();
+
+    char[] chars = value.toCharArray();
+    boolean titleCased = false;
+
+    for (int i = 0; i < chars.length; i++) {
+      if (chars[i] == ' ') {
```

Comment:
I think you want to use `Character.isWhitespace(chars[i])` to handle `\n`, `\t`, etc.

Reply:
Unfortunate timing @mattcoley . Got merged while I was typing this comment

Reply:
Good catch, I reverted the PR and will add some test cases for tabs and new lines.